File: model/book.dart
class book{
  int id=0;
  String name="";
  String Author="";
  dynamic Quantity=0;
  dynamic Price=0;
  String Category="";

  

  Map<String,dynamic> mapbook(){
    Map<String,dynamic> m={
      "id":this.id ,
      "name":this.name,
      "category":this.Category,
      "author":this.Author,
      "quantity":this.Quantity ,
      "price":this.Price ,
    };

return m;
  }
}

File: app/Models/Book.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Book extends Model
{
    use HasFactory;

    protected $fillable = [
        "name",
        "published_at"
    ];

    public function category()
    {
        
        return $this->belongsTo(Category::class,);
    }

    public function authors()
    {
        
        return $this->belongsToMany(Author::class, 'author_book')->withPivot('is_active')->withTimestamps();
    }
}
